ui/job-view/pushes/Push.jsx

```diff
@@ -128,8 +128,12 @@ class Push extends React.Component {
   }
 
   mapPushJobs(jobs, skipJobMap) {
+    const { updateJobMap, recalculateUnclassifiedCounts, push } = this.props;
+
+    // whether or not we got any jobs for this push, the operation to fetch
+    // them has completed.
+    push.jobsLoaded = true;
```

Comment:
This was the critical line for the bug.  In some cases we may get a push that has no jobs.  Super unlikely, but I saw it in this case.  So we didn't mark ``jobsLoaded`` as true.  And we only try to select the job in question down lower when all pushes have ``jobsLoaded`` true.